api/src/main/java/org/openmrs/module/queue/api/dao/impl/AbstractBaseQueueDaoImpl.java

```diff
@@ -115,4 +119,15 @@ protected void includeVoidedObjects(Criteria criteria, boolean includeRetired) {
 			}
 		}
 	}
+	
+	/**
+	 * Creates concept names subQuery
+	 *
+	 * @param conceptName Concept name
+	 * @return {@link DetachedCriteria} conceptName subQuery
+	 */
+	protected DetachedCriteria conceptByNameDetachedCriteria(@NotNull String conceptName) {
+		return DetachedCriteria.forClass(ConceptName.class, "cn").add(Restrictions.eq("cn.name", conceptName))
```

Comment:
We probably want to restrict this to either fully-specified names or locale-preferred names. Otherwise, this could match more than one concept.

Reply:
Yes. 💯